src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,632 @@
+                <orgX:nameX:versionX>,
+                <orgY:nameY:versionY>,
+              ]
+            },
+            {
+              "coord": "orgB:nameB:versionB",
+              "files": [
+                [
+                  <classifier>,
+                  <path>,
+                ]
+              ],
+              "dependencies": [ // coodinates for its transitive dependencies
+                <orgX:nameX:versionX>,
+                <orgZ:nameZ:versionZ>,
+              ]
+            },
+            ... // more about orgX:nameX:versionX, orgY:nameY:versionY, orgZ:nameZ:versionZ
+          ]
+        }
+    Hence the aggregation of the results will be in the following format, for example when default classifier
+    and sources are fetched:
+    {
+      '': <result from first coursier call with default classifier>,
+      'sources': <result from second coursier call with 'sources' classifier>,
+    }
+    """
+
+    # Prepare coursier args
+    coursier_subsystem_instance = CoursierSubsystem.global_instance()
+    coursier_jar = coursier_subsystem_instance.bootstrap_coursier(self.context.new_workunit)
+
+    common_args = ['fetch',
+                   # Print the resolution tree
+                   '-t',
+                   '--cache', coursier_cache_path
+                   ] + coursier_subsystem_instance.get_options().fetch_options
+
+    def construct_classifier_to_jar(jars):
+      """
+      Reshape jars by classifier
+
+      For example:
+      [
+        ResolvedJar('sources', 'a-sources.jar'),
+        ResolvedJar('sources', 'b-sources.jar'),
+        ResolvedJar('', 'a.jar'),
+      ]
+      should yield:
+      {
+        'sources': [ResolvedJar('sources', 'a-sources.jar'),ResolvedJar('sources', 'b-sources.jar')],
+        '':  [ResolvedJar('', 'a.jar') ]
+      }
+      """
+      product = defaultdict(list)
+      for x in jars:
+        product[x.coordinate.classifier or ''].append(x)
+      return product
+
+    # Divide the resolve by classifier because coursier treats classifier option globally.
+    classifier_to_jars = construct_classifier_to_jar(jars_to_resolve)
+
+    # Variable to store coursier each run.
+    results = {}
+
+    for classifier, classified_jars in classifier_to_jars.items():
+
+      coursier_work_temp_dir = os.path.join(pants_workdir, 'tmp')
+      safe_mkdir(coursier_work_temp_dir)
+      with temporary_file(coursier_work_temp_dir, cleanup=False) as f:
+        output_fn = f.name
+        common_args.extend(['--json-output-file', output_fn])
+
+      cmd_args = self._construct_cmd_args(classified_jars, classifier, common_args,
+                                          global_excludes if self.get_options().allow_global_excludes else [],
+                                          pinned_coords, coursier_work_temp_dir)
+
+      with self.context.new_workunit(name='coursier', labels=[WorkUnitLabel.TOOL]) as workunit:
```

Comment:
Note: Just filed https://github.com/pantsbuild/pants/issues/5395 re this superfluous workunit.